File: Lab 4/Lab 4 c++/Funcionario.cpp

#include "Funcionario.h"

Funcionario::Funcionario(const std::string& nome, int salario) : nome(nome), salario(salario) {}

std::string Funcionario::getNome() const {
    return nome;
}

void Funcionario::setNome(const std::string& nome) {
    this->nome = nome;
}

int Funcionario::getSalario() const {
    return salario;
}

void Funcionario::setSalario(int salario) {
    this->salario = salario;
}

std::string Funcionario::descricao() const {
    return "Nome: " + nome + "\nSalario: " + std::to_string(salario);
}

File: Lab 4/Lab 4 Java/Funcionario.java

public class Funcionario {
    protected String nome;
    protected int salario;

    public Funcionario(String nome, int salario) {
        this.nome = nome;
        this.salario = salario;
    }

    public String descricao() {
        return "Nome: " + nome + "\nSalario: " + salario;
    }
}
